### ISSUE TITLE
Docs CI frequently fails due to sporadic link checking issues

### ISSUE THREAD

**Issue by jchodera** [openmm/openmm#4123]
I was surprised to find that many of the recent merged PRs had successfully completed all tests:
![image](https://github.com/openmm/openmm/assets/3656088/ac78b703-ce92-4f5e-822f-cae0763b86c7)
One reason for this appears to be that the docs tests are failing. Here's an example:
https://github.com/openmm/openmm/actions/runs/5326129157/jobs/9647743715

The "Check 404s" appears to have identified a broken URL:
```
 Check userguide
  🏊‍♂️ crawling ./userguide/
  [0] https://www.latex-project.org/get
  userguide/library/02_compiling.html
    [0] https://www.latex-project.org/get
  ERROR: Detected 1 broken links. Scanned 117 links in 4.062 seconds.
 Check README
Error: Process completed with exit code 1.
```
This link appears to be fine, however. Perhaps we simply need to pass additional flags to the [link checker](https://github.com/JustinBeckwith/linkinator) to have it retry on error? Here are the relevant flags:
```
    --retry,
        Automatically retry requests that return HTTP 429 responses and include
        a 'retry-after' header. Defaults to false.

    --retry-errors,
        Automatically retry requests that return 5xx or unknown response.

    --retry-errors-count,
        How many times should an error be retried?

    --retry-errors-jitter,
        Random jitter applied to error retry.
```
The link checker is invoked [here](https://github.com/openmm/openmm/blob/master/.github/workflows/CI.yml#L682-L687) in the CI.

**Comment by peastman** [comment]
Yes, the docs build sometimes fails for transient issues.  That particular one has fixed itself in the most recent builds.

Also, the Jenkins build always fails due to #4109.  If anyone has suggestions about the correct fix for it, that would be great.
